test/test_launcher.rb

```diff
@@ -79,6 +79,57 @@ def test_pid_file
     File.unlink tmp_path
   end
 
+  def test_state_permission_0640
+    tmp_file = Tempfile.new("puma-state")
+    tmp_path = tmp_file.path
+    tmp_file.close!
+    tmp_permission = 0640
+
+    conf = Puma::Configuration.new do |c|
+      c.state_path tmp_path
+      c.state_permission tmp_permission
+    end
+
+    launcher(conf).write_state
+
+    assert File.stat(tmp_path).mode.to_s(8)[-4..-1], tmp_permission
+
+    File.unlink tmp_path
```

Comment:
You can make this a bit more guaranteed to run:
```suggestion
ensure
    File.unlink tmp_path if tmp_path
```

Reply:
great suggestion, added now.